src/custom/pages/Swap/index.tsx

```diff
@@ -124,6 +126,67 @@ const SwapModWrapper = styled(SwapMod)`
   }
 `
 
+const fadeIn = `
+  @keyframes fadeIn {
+    0% {
+      opacity: 0;
+    }
+    100% {
+      opacity: 1;
+    }
+  }
+`
+
+interface LoadingCowImgProps {
+  maxWidth?: string
+  showLoader: boolean
+}
+
+const LoadingCowImg = styled.img<LoadingCowImgProps>`
+  max-width: ${({ maxWidth = '60px' }) => maxWidth};
+  opacity: ${({ showLoader }) => (showLoader ? 1 : 0)};
+
+  transition: opacity 0.4s ease-in-out;
+`
+
+type LoadingCowProps = {
+  isSoftLoading: boolean
+} & { maxWidth?: string; src: string }
+
+const LOADING_COW_TIMER = 4000
+
+function LoadingCow(props: LoadingCowProps) {
+  const { isSoftLoading, ...rest } = props
+  const [showCow, setShowCow] = useState(isSoftLoading)
+
+  useEffect(() => {
+    let timeout: NodeJS.Timeout | undefined
+    // price is being queried
+    if (isSoftLoading) {
+      // isn't currently showing cow or hasnt yet been updated here
+      // so we clear any running timeouts ready to clear local loading state
+      // and essentially reset them
+      clearTimeout(timeout as NodeJS.Timeout)
+      setShowCow(true)
+    } else {
+      // no longer loading
+      // reset timeout to clear local loading state after LOADING_COW_TIMER ms
+      if (showCow) {
+        timeout = setTimeout(() => {
```

Comment:
nit: you can count the time that has been shown, and if it's less than the minimum animation time, you hide after the time difference

```js
const animationStart = Date.now()
...

if (showCow) {
    const timeRemainingAnimation = Math.max(0, LOADING_COW_TIMER - Date.now() - animationStart)
    setTimeout(hideCow, timeRemainingAnimation)
}

```

Reply:
tagged in issue #752